Extend configurability of dxp's position on screen

mmskv opened the pull request:
- dxp can be placed in any of the moniotr's corners
- Specify what monitor to put dxp on with `dxp_monitor_name`
- `dxp_monitor` defaults to primary monitor with `get_monitor_primary()`
- dxp can be centered both vertically and horizontally

Closes #14

mmskv committed:
Add more configurations for dxp position

Monitor for the dxp to be drawn on can be specified.
dxp can be placed in any of the four corners of the monitor.

WIP centering on one axis.

mmskv committed:
Add dxp centering config option

mmskv committed:
Unset dxp_monitor_name in default config

TheHussky commented on src/window.cpp:
```diff
@@ -2,16 +2,16 @@
-                const std::vector<dxp_socket_desktop> &desktops)
-    : drawable (x, y, 0, 0) // Width and height will be calculated from config
+window::window (const std::vector<dxp_socket_desktop> &desktops)
+    : drawable (0, 0, 0, 0) // Width and height will be calculated from config
```
Do we really need first two arguments then? I think there's a better way to code it.

TheHussky replied:
oh, the function requires it, i see. Still, those four zeros seem weird. Maybe set args by default?

mmskv replied:
will do

TheHussky commented on src/xcb_util.hpp:
```diff
@@ -80,6 +81,13 @@ std::vector<uint32_t> get_property_value (xcb_connection_t *c,
+/**
+ * Get crtc of the primary monitor.
+ *
+ * NOTE: Returned monitor_info.name is unsed as it is unused.
```
it's unset (not unsed) i believe. I guess it's better rto fix it here and in a couple more places

mmskv replied:
wdym?

mmskv replied:
fixed

TheHussky reviewed (approved):
You may want to fix minor inconveniences

mmskv committed:
Fix typo in docs

mmskv committed:
Add default constructor to drawable

Default constructor sets x, y, width, height to zero.
This is used in window constructor as these params are unknown at first

TheHussky reviewed (approved):
lgtm
